### DISCUSSION TITLE
How to create button which shows state select/not selected

### DISCUSSION THREAD

**Discussion by chess-levin** [TomSchimansky/CustomTkinter#2328] [General]
I'd like to visualize the selection state of the buttons in my vlc-interface-app. The user can select on favorite-list and I'd like to show which one is active. It's more less the logic of a set of radio-buttons. Push on button and the other button will be unselected, but I need big buttons, because it runs on a touch display.

![image](https://github.com/TomSchimansky/CustomTkinter/assets/6573503/8f677be7-e602-4533-bd55-32f3b7af9230)

I've started some code by extending `CTKButton`, but I don't know how/where to intercept the button pushed event. Or would it be easier to use a [CTkRadioButton](https://customtkinter.tomschimansky.com/documentation/widgets/radiobutton)  and replace the selected circle by changing the background color of the radio button?

```
class StateButton(ctki.CTkButton):
    def __init__(self, master, selected_color):
        super().__init__(master)

        self.state = False
        self.unselected_color = self._fg_color
        self.selected_color = selected_color

        self._command

    def select(self):
        self.state = True

    def unselect(self):
        self.state = False

    def viewState(self):
        if self.state:
            self.configure(fg_color=self.selected_color)
        else:
            self.configure(fg_color=self.unselected_color)
```

**Comment by Anonymous6598** [comment]
It would be easier to use CTkRadioButton.

**Comment by Intiiii** [comment]
Hi, i'm kinda new to programming, but wouldn't it be possible to create like a command that, after a button is pressed, changes the color value the button, or maybe changes the image to smth different? 

Then it would look a little like this

def button_pressed():
      button_color = {color of active button}

but then again, i'm really new, but this is how I would expect it to look like, but idk how it works with colors and buttons with CTk...
